lib/client/indexer/market-clients/ammMarkets.ts

```diff
@@ -24,55 +26,72 @@ export class FutarchyIndexerAmmMarketsClient
   ): Promise<AmmMarket | undefined> {
     return this.rpcMarketsClient.fetchMarket(request);
   }
+
+  validateAddLiquidity(
+    ammMarket: AmmMarket,
+    quoteAmount: number,
+    baseAmount: number
+  ): LiquidityAddError | null {
+    return this.rpcMarketsClient.validateAddLiquidity(
+      ammMarket,
+      quoteAmount,
+      baseAmount
+    );
+  }
+
   async addLiquidity(
-    ammAddr: PublicKey,
-    quoteAmount?: number | undefined,
-    baseAmount?: number | undefined
-  ): Promise<string[]> {
-    return this.rpcMarketsClient.addLiquidity(ammAddr, quoteAmount, baseAmount);
+    ammMarket: AmmMarket,
+    quoteAmount: number,
+    baseAmount: number
+  ): Promise<string[] | LiquidityAddError> {
+    return this.rpcMarketsClient.addLiquidity(
+      ammMarket,
+      quoteAmount,
+      baseAmount
+    );
   }
 
   async simulateAddLiquidity(
-    ammAddr: PublicKey,
+    ammMarket: AmmMarket,
     baseAmount: number,
     quoteAmount: number
   ): Promise<any> {
     // Replace `any` with the actual return type if defined
     return this.rpcMarketsClient.simulateAddLiquidity(
-      ammAddr,
+      ammMarket,
       baseAmount,
       quoteAmount
     );
   }
 
   async removeLiquidity(
-    ammAddr: PublicKey,
+    ammMarket: AmmMarket,
     lpTokensToBurn: number
   ): Promise<string[]> {
-    return this.rpcMarketsClient.removeLiquidity(ammAddr, lpTokensToBurn);
+    return this.rpcMarketsClient.removeLiquidity(ammMarket, lpTokensToBurn);
   }
 
   async swap(
-    ammAddr: PublicKey,
+    ammMarket: AmmMarket,
     swapType: SwapType,
     inputAmount: number,
     outputAmountMin: number
   ): Promise<string[]> {
     return this.rpcMarketsClient.swap(
-      ammAddr,
+      ammMarket,
       swapType,
       inputAmount,
       outputAmountMin
```

Comment:
Just as a reference, this needs to be calculated on the UI side OR / AND we can pass in a slippage param to the SDK such that we validate it in both areas... 

You output amount is the `estimated - (estimated * slippage)`

Reply:
`outputAmountMin` that value should be the result of  what we have **multiplied** by `(1 - slippage)`

Reply:
This has been fixed now in here: https://github.com/metaDAOproject/metadao-frontend/pull/56/commits/22de20d4c41fffc159fad43796d805aeacb8232e